signer/src/stacks/api.rs

```diff
@@ -388,20 +388,28 @@ impl TenureBlockHeaders {
     }
 
     /// Get the minimum block height in the tenure.
-    pub fn start_height(&self) -> StacksBlockHeight {
+    pub fn start_header(&self) -> StacksBlockHeader {
         // SAFETY: It is okay to unwrap here because we know that the
         // tenure is non-empty. The struct upholds this invariant upon
         // creation.
-        self.headers.iter().map(|h| h.block_height).min().unwrap()
+        self.headers
+            .iter()
+            .min_by_key(|header| header.block_height)
+            .unwrap()
+            .clone()
```

Comment:
We don't need the `.clone()`s here and in `end_header`, we only need the references.